pkg/landscaper/controllers/deployitem/controller.go

```diff
@@ -75,35 +71,171 @@ func (con *controller) Reconcile(ctx context.Context, req reconcile.Request) (re
 		return reconcile.Result{}, err
 	}
 
+	var requeue *time.Duration
+	var err error
+	old := di.DeepCopy()
+
+	// detect pickup timeout
+	if con.pickupTimeout != 0 {
+		logger.V(7).Info("check for pickup timeout")
+		requeue, err = con.detectPickupTimeouts(logger, di)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		if !reflect.DeepEqual(old.Status, di.Status) {
+			if err := con.c.Status().Update(ctx, di); err != nil {
+				logger.Error(err, "unable to set deployitem status")
+				return reconcile.Result{}, err
+			}
+			// if there was a pickup timeout, no need to check for anything else
+			return reconcile.Result{}, nil
+		}
+	}
+
+	// detect aborting timeout
+	if con.abortingTimeout != 0 {
+		logger.V(7).Info("check for aborting timeout")
+		tmp, err := con.detectAbortingTimeouts(logger, di)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		if requeue == nil {
+			requeue = tmp
+		} else if tmp != nil && *tmp < *requeue {
+			requeue = tmp
+		}
+		if !reflect.DeepEqual(old.Status, di.Status) {
+			if err := con.c.Status().Update(ctx, di); err != nil {
+				logger.Error(err, "unable to set deployitem status")
```

Comment:
can you add a comment that we might have to expose such errors as events on the deploy item